modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java

```diff
@@ -1975,6 +1979,54 @@ public void testVisorGateway() throws Exception {
         jsonTaskResult(ret);
     }
 
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testDataRegionMetrics() throws Exception {
+        String ret = content(F.asMap("cmd", GridRestCommand.DATA_REGION_METRICS.key()));
+
+        JsonNode res = validateJsonResponse(ret);
+
+        assertTrue(res.size() > 0);
+
+        info(GridRestCommand.DATA_REGION_METRICS.key().toUpperCase() + " command result: " + ret);
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testDataStorageMetricsDisabled() throws Exception {
+        String ret = content(F.asMap("cmd", GridRestCommand.DATA_STORAGE_METRICS.key()));
+
+        JsonNode res = validateJsonResponse(ret);
+
+        assertTrue(res.asText().equalsIgnoreCase("Storage metrics are not enabled"));
+
+        info(GridRestCommand.DATA_STORAGE_METRICS.key().toUpperCase() + " command result: " + ret);
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testDataStorageMetricsEnabled() throws Exception {
+        stopAllGrids();
+
+        memoryMetricsEnabled = true;
```

Comment:
Test execution order is not defined usually.
Let's add additional check and restart the grid if this option has to be changed.
Same change need to be done in testDataStorageMetricsDisabled().